app/Http/Controllers/NotesController.php

```diff
@@ -3,18 +3,125 @@
+     * @param NoteService $noteService
      * @return View
      */
-    public function overlay(Request $request): View
+    public function overlay(Request $request, NoteService $noteService): View
     {
-        return view('ingame.notes.overlay');
+        $data = [];
+
+        $notesDeleted = false;
+
+        if ($request->isMethod('post')) {
+            $deleteMethod = $request->input('noticeDeleteMethode');
+
+            if ($deleteMethod == 1) {
+                $noteIds = $request->input('delIds', []);
+                if (!empty($noteIds)) {
+                    $noteService->deleteMarkedNotes($noteIds);
+                    $notesDeleted = true;
+                }
+            } elseif ($deleteMethod == 2) {
+                $noteService->deleteAllNotesForUser();
+                $notesDeleted = true;
+            }
+
+            if ($notesDeleted) {
+                $data['success'] = __('Note(s) has(ve) been deleted');
+            }
+        }
+
+        $data['notes'] = $noteService->getAllNotesForUser();
+        return view('ingame.notes.overlay', $data);
+    }
+
+    /**
+     * Shows the notes view popup page
+     *
+     * @param Request $request
+     * @param NoteService $noteService
+     * @return View
+     */
+    public function view(Request $request, NoteService $noteService): View
+    {
+        $note = null;
+
+        $noteId = $request->input('id');
+        if ($noteId) {
+            $note = $noteService->getNoteById($noteId);
+        }
+
+        return view('ingame.notes.create')->with([
+            'noteId' => $note ? $note->id : 0,
+            'priority' => $note ? $note->priority : 2,
+            'subject' => $note ? $note->subject : '',
+            'content' => $note ? $note->content : '',
+        ]);
+    }
+
+    /**
+     * Create a new note
+     *
+     * @param Request $request
+     * @param NoteService $noteService
+     * @return JsonResponse
+     */
+    public function ajaxCreate(Request $request, NoteService $noteService): JsonResponse
+    {
+        $validated = $request->validate([
+            'id' => [
+                'nullable',
+                'integer',
+                function ($attribute, $value, $fail) use ($noteService) {
+                    if ($value !== '0' && $value !== null && !$noteService->noteExistsAndBelongsToUser($value)) {
+                        $fail('The selected note ID does not exist.');
+                    }
+                },
+            ],
+            'noticePrio' => 'nullable|integer|min:1|max:3',
+            'noticeSubject' => 'nullable|string|max:32',
+            'noticeText' => 'nullable|string|max:5000',
+        ]);
+
+        $data = [
+            'priority' => $validated['noticePrio'] ?? 2,
+            'subject' => $validated['noticeSubject'] ?? __('Your subject'),
+            'content' => $validated['noticeText'],
+        ];
+
+        try {
+            if (isset($validated['id']) && $validated['id'] != 0) {
```

Comment:
Same as mentioned above: you could use the PHP `empty()` check here which will do the same as both existing checks.

Reply:
Indeed you are totally right, i started with the first check and added the second check later and omitted to convert it.